source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -69,16 +72,23 @@ public class WriteTransaction : ReadTransaction, IRelationalTransaction
         {
             IReadOnlyList<object> documentList = documents.ToArray();
             if (!documentList.Any()) return;
+            
+            var batchBlockSize = BatchBlockSize(documentList, options);
+            if (batchBlockSize == 0) throw new InvalidOperationException("Document has exceeded the supported value of 2100 parameters for a single row");
 
-            var command = builder.PrepareInsert(documentList, options);
-            foreach (var document in documents) configuration.Hooks.BeforeInsert(document, command.Mapping, this);
+            foreach (var documentsBatch in documentList.BatchWithBlockSize(batchBlockSize))
+            {
+                var documentsToInsert = documentsBatch.ToArray();
+                var command = builder.PrepareInsert(documentsToInsert, options);
+                foreach (var document in documentsToInsert) configuration.Hooks.BeforeInsert(document, command.Mapping, this);
 
-            var outputs = ExecuteDataModification(command);
-            ApplyNewRowVersionsIfRequired(documentList, command.Mapping, outputs);
-            ApplyIdentityIdsIfRequired(documentList, command.Mapping, outputs);
+                var outputs = ExecuteDataModification(command);
```

Comment:
Lots of noise on the diff. Essentially adding indentation from the introduction of foreach